supervisor/api/network.py

```diff
@@ -300,6 +311,16 @@ async def create_vlan(self, request: web.Request) -> None:
 
         vlan_config = VlanConfig(vlan, interface.name)
 
+        # Set MDNS mode for VLANs to RESOLVE by default
+        mdns_mode = MulticastDnsMode.RESOLVE
+        llmnr_mode = MulticastDnsMode.RESOLVE
```

Comment:
This essentially makes mDNS and LLMNR unusable for those of us who respect the intention of link-local protocols (which is, to operate them link local only, see also https://github.com/home-assistant/supervisor/issues/5435#issuecomment-2500411115).

For those who run reflector it should be easy to switch the switch. But let's go with default configuration by default.

```suggestion
        mdns_mode = MulticastDnsMode.DEFAULT
        llmnr_mode = MulticastDnsMode.DEFAULT
```